core/src/main/java/com/google/instrumentation/trace/TraceExporter.java

```diff
@@ -54,6 +75,393 @@
+     */
+    public abstract Summary getSummary();
+
+    /**
+     * Returns a list of active spans that are matching the {@code filter}.
+     *
+     * @param filter used to filter the returned spans.
+     * @return a list of active spans that are matching the {@code filter}.
+     */
+    public abstract List<SpanData> getActiveSpans(ActiveSpansFilter filter);
+
+    /**
+     * Returns a list of succeeded spans (spans with {@link Status} equal to {@link Status#OK}) that
+     * are matching the {@code filter}.
+     *
+     * <p>Latency based sampled spans are available only for span names registered using {@link
+     * #collectSamplesForSpanNames(List)}.
+     *
+     * @param filter used to filter the returned sampled spans.
+     * @return a list of succeeded spans that are matching the {@code filter}.
+     */
+    public abstract List<SpanData> getLatencyBasedSampledSpans(
+        LatencyBasedSampledSpansFilter filter);
+
+    /**
+     * Returns a list of failed spans (spans with {@link Status} different than {@link Status#OK})
+     * that are matching the {@code filter}.
+     *
+     * <p>Error based sampled spans are available only for span names registered using {@link
+     * #collectSamplesForSpanNames(List)}.
+     *
+     * @param filter used to filter the returned sampled spans.
+     * @return a list of failed spans that are matching the {@code filter}.
+     */
+    public abstract List<SpanData> getErrorBasedSampledSpans(ErrorBasedSampledSpansFilter filter);
+
+    /**
+     * Registers a list of span names for which the library will collect latency based sampled spans
+     * and error based sampled spans.
+     *
+     * @param spanNames list of span names for which the library will collect samples.
+     */
+    public abstract void collectSamplesForSpanNames(List<String> spanNames);
+
+    /** The summary of all in-process debugging information. */
+    @AutoValue
+    @Immutable
+    public abstract static class Summary {
+
+      Summary() {}
+
+      /**
+       * Returns a new instance of {@code Summary}.
+       *
+       * @param perSpanNameSummary a map with summary for each different span name.
+       * @return a new instance of {@code Summary}.
+       * @throws NullPointerException if {@code perSpanNameSummary} is {@code null}.
+       */
+      public static Summary create(Map<String, PerSpanNameSummary> perSpanNameSummary) {
+        return new AutoValue_TraceExporter_InProcessDebuggingHandler_Summary(
+            Collections.unmodifiableMap(
+                new HashMap<String, PerSpanNameSummary>(
+                    checkNotNull(perSpanNameSummary, "perSpanNameSummary"))));
+      }
+
+      /**
+       * Returns a map with summary of available data for each different span name.
+       *
+       * @return a map with all the span names and the summary.
+       */
+      public abstract Map<String, PerSpanNameSummary> getPerSpanNameSummary();
```

Comment:
It looks like a lot of tracing APIs use span name. It might be worth adding a `SpanName` class to make it clear which ones refer to the same thing. For example, `SpanData` has a `displayName` and `SpanImpl` has a `name` field. I'm not suggesting making the change in this PR, though.